vortex-btrblocks/src/float.rs

```diff
@@ -201,7 +202,8 @@ impl Scheme for ALPScheme {
         allowed_cascading: usize,
         excludes: &[FloatCode],
     ) -> VortexResult<ArrayRef> {
-        let alp = alp_encode(stats.source())?;
+        let alp_encoded = ALPEncoding.encode(&stats.source().to_canonical()?, None)?;
+        let alp = alp_encoded.as_::<ALPArray>();
```

Comment:
we can still reach for `alp_encode` here, but this seems like the "right" thing to do, and its still pretty ugly IMO.
I also don't like that we need to downcast right after it, but given that `EncodingVTable` needs to be dyn-compatible I'm not sure we can do better here.